src/main/ruby/truffleruby/core/string.rb

```diff
@@ -70,6 +70,54 @@ def byteslice(index_or_range, length = undefined)
     byteslice index, length
   end
 
+  def bytesplice(*args)
```

Comment:
I think this could be `def bytesplice(index_or_range, length = undefined, str)` (post-required args are nice here).
In fact `byteslice` already uses a similar signature.
Then no need to deal with this as an array and it's much easier to read and know what this method takes.

Reply:
Ah yes, that looks nicer. Updating.